front/lib/api/assistant/actions/visualization.ts

```diff
@@ -51,18 +52,20 @@ export class VisualizationAction extends BaseAction {
     super(blob.id, "visualization_action");
 
     this.agentMessageId = blob.agentMessageId;
-    this.query = blob.query;
     this.output = blob.output;
     this.functionCallId = blob.functionCallId;
     this.functionCallName = blob.functionCallName;
     this.step = blob.step;
   }
 
+  // Visualization is not a function call, it is pure generation cause we need streaming.
+  // We fake a function call for the multi-actions model because
+  // we cannot render two agent messages in a row.
   renderForFunctionCall(): FunctionCallType {
     return {
       id: this.functionCallId ?? `call_${this.id.toString()}`,
```

Comment:
Just in case you don't have this, I think we should grab the function call ID from the `runMultiActionsAgent`

Reply:
Yes that's what I did!